cookbook/teaser-provider.rst

```diff
@@ -1,121 +0,0 @@
-Provider for Teaser-Selection
```

Comment:
I think that makes sense so that it's easier to find that. I also restructured the sections a bit to improve the flow when reading the guide:

* Introductory text to explain what a content type is about. Can be very short (e.g. for text) or longer (for content types that are harder to understand)
* Section "Configuration" explains the *easiest possible* configuration
* Section "Rendering" explains the *easiest possible* rendering, including the properties in Twig
* Section "Parameters" provides a reference about all the parameters
* Additional sections explain alternative usages in detail, with code examples and images (if necessary)